components/camel-ipfs/src/main/java/org/apache/camel/component/ipfs/IPFSComponent.java

```diff
@@ -54,6 +59,11 @@ protected Endpoint createEndpoint(String urispec, String remaining, Map<String,
         config.setIpfsCmd(cmd);
 
         client = createClient(config);
+        try {
```

Comment:
See JIRA comment about not having such logic here, but it belongs in doStart/doStop etc.